Automattic/jetpack
# Forms: Ensure form choice buttons have a fallback text color if text and background are similar

Fixes https://github.com/Automattic/vulcan/issues/539

## Proposed changes:

* This PR checks if the form button text color is similar to the background color, and if so sets the background color to the general button default. This fixes an issue on multiple themes where multi-choice and single-choice form options (when using the button style), had similar text and button background color.
  * There are existing discrepancies between editor and front-end across themes, noted in the original PR here - https://github.com/Automattic/jetpack/pull/30319#issuecomment-1528034626 - but the main aim of this PR is ensure button text is readable on the front-end where-ever possible.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/539

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Self-hosted:

* On trunk, add a form and add both multi-choice and single choice field options to a post or page. Test the following with the Twenty Twenty, Astra, Twenty Fifteen and Twenty Fourteen themes:
  * On the front-end of the site, notice that the text and button color is the same, unless you deliberately highlight the text. Astra is slightly different in that the single choice options do show a text color.
* Apply this PR, and test with the same themes. Notice each now show the same default text color as you see in other form buttons on the post / page.
* If you can, test with any other themes as well.
* There are existing discrepancies between editor and front-end across themes, mentioned under 'Proposed changes' above, so this is expected at the moment.

WoA:

* As with self-hosted.

Simple:

* As with self-hosted (using the command in the generated comment below, for a sandboxed Simple site). Note that Astra is not available for Simple sites.
* There are additional themes that had broken button styling on the front-end, only on Simple, that are fixed by this PR as well: Dara and Sketch.



**Screenshots to demo the difference**

Twenty Twenty before:

<img width="250" alt="Screenshot 2024-10-17 at 13 50 00" src="https://github.com/user-attachments/assets/4a9e870b-56af-4154-86b0-a7cc63f4940e">


Twenty Twenty after:
<img width="246" alt="Screenshot 2024-10-17 at 13 50 09" src="https://github.com/user-attachments/assets/0878116f-5042-4afc-8178-ed3a420de057">
